streams/src/test/java/org/apache/kafka/streams/integration/KStreamKTableJoinIntegrationTest.java

```diff
@@ -251,10 +258,23 @@ public RegionWithClicks apply(final Long clicks, final String region) {
                 public Long apply(final Long value1, final Long value2) {
                     return value1 + value2;
                 }
-            }, "ClicksPerRegionUnwindowed");
+            });
 
         // Write the (continuously updating) results to the output topic.
-        clicksPerRegion.to(stringSerde, longSerde, outputTopic);
+        clicksPerRegion
+            .toStream()
+            .peek(new ForeachAction<String, Long>() {
+                @Override
+                public void apply(final String key, final Long value) {
```

Comment:
I'm not sure about this part, I understand the reasoning behind it, but it feels a bit like we are "fudging" a little for the test to pass.  Maybe we could assert a minimum of expected values has been processed or something along those lines?

Reply:
Don't we apply a minimum.
 1. `remainingExpectedResult` must be empty -- thus, we expect to see the final correct result here (cf. https://github.com/apache/kafka/pull/4364/files#diff-bc785804f37105362649cb12829f76aaR320)
 2. we expect at minimum of output records (cf. https://github.com/apache/kafka/pull/4364/files#diff-bc785804f37105362649cb12829f76aaR327)

Reply:
Ack, thanks for clarifying.  FWIW I ran 100x on two separate occasions and all passed.